docs/contributor/03-55-zones-discovery.md

```diff
@@ -0,0 +1,39 @@
+# Zones Discovery
+
+## Overview
+
+The Zones Discovery feature extends Kyma Environment Broker (KEB) to dynamically resolve availability zones for Kyma and additional worker node pools during provisioning and updates.
+Operators can configure worker node pools to use either static zone assignments (predefined in configuration) or dynamic zone assignments (queried live from the hyperscaler).
+
+> [!NOTE]
+> This feature is currently supported only on AWS.
+
+Configuration:
+
+```yaml
+providersConfiguration:
+  aws:
+    zonesDiscovery: true
+```
+
+If both a static configuration and `zonesDiscovery` are provided, a warning is logged to indicate that static zones are ignored.
+
+## Validation
+
+During provisioning and updates, KEB validates worker node pool configuration by retrieving a random hyperscaler subscription secret from Gardener and using it to query the available zones for the specified machine type.
+- The Kyma worker node pool must support at least three zones.
+- Additional worker pools must support three zones if configured for high availability, or at least one zone otherwise.
+
+The subscription secret is used only for validation. Its name is logged to support traceability in case of validation failures.
+
+## Zones Discovery
+
+If `zonesDiscovery` is enabled, KEB performs the `Discover_Available_Zones` step using hyperscaler credentials from the subscription secret resolved in the `Resolve_Subscription_Secret` step.
+For each unique machine type across the Kyma and additional worker node pools (during provisioning) or additional worker node pools (during updates), KEB queries all available zones.
```

Comment:
During provisioning, KEB queries all available zones for each unique machine type across both the Kyma worker node pool and additional worker node pools. During updates, it queries zones only for the additional worker node pools.